src/renderer/src/hooks/useGetFullTransactions.tsx

```diff
@@ -0,0 +1,316 @@
+        allAccounts,
+        networksByBlockchain,
+        page,
+      }),
+    initialPageParam: 1,
+    getNextPageParam: ({ page }) => page,
+  })
+
+  const data = useMemo(() => {
+    const filteredPendingTransfersByAccounts = pendingTransactions.filter(
+      ({ account, time }) =>
+        accounts.some(AccountHelper.predicate(account)) &&
+        isWithinInterval(createNewDateByTime(time), { start: dateFrom, end: dateTo })
+    )
+
+    const groupedPendingTransfersByHash = groupBy(filteredPendingTransfersByAccounts, 'hash')
+    const items = query.data?.pages?.flatMap(page => page?.data ?? []) ?? []
+
+    const filteredItems = items.filter(
+      ({ txId }, index, array) =>
+        !groupedPendingTransfersByHash[txId] && array.findIndex(item => item.txId === txId) === index
+    )
+
+    filteredItems.unshift(
+      ...Object.values(groupedPendingTransfersByHash).map(transfers => {
+        transfers.sort(({ time: newerTime }, { time: olderTime }) => newerTime - olderTime)
+
+        const [firstTransfer] = transfers
+
+        const {
+          hash,
+          account: { blockchain },
+        } = firstTransfer
+
+        const transaction = items.find(({ txId }) => txId === hash)
+        const normalizedHiddenTokenHashes = hiddenTokensByBlockchain[blockchain]?.map(UtilsHelper.normalizeHash) ?? []
+        const service = bsAggregator.blockchainServicesByName[blockchain]
+
+        let txTemplateUrl: string | undefined
+        let addressTemplateUrl: string | undefined
+        let contractTemplateUrl: string | undefined
+
+        if (hasExplorerService(service)) {
+          txTemplateUrl = service.explorerService.getTxTemplateUrl()
+          addressTemplateUrl = service.explorerService.getAddressTemplateUrl()
+          contractTemplateUrl = service.explorerService.getContractTemplateUrl()
+        }
+
+        return {
+          txId: hash,
+          txIdUrl: hash ? txTemplateUrl?.replace('{txId}', hash) : undefined,
+          block: transaction?.block || 0,
+          date: createNewDateByTime(firstTransfer.time).toISOString(),
+          invocationCount: transaction?.invocationCount || 0,
+          notificationCount: transaction?.notificationCount || 0,
+          networkFeeAmount: transaction?.networkFeeAmount || '0',
+          systemFeeAmount: transaction?.systemFeeAmount || '0',
+          blockchain,
+          isPending: true,
+          events:
+            transaction?.events ||
+            transfers
+              .map(transfer => {
+                const normalizedHash = UtilsHelper.normalizeHash(transfer.assetHash)
+                const lowerCaseSymbol = transfer.asset.toLowerCase()
+
+                const token = service.tokens.find(
+                  ({ hash, symbol }) =>
+                    UtilsHelper.normalizeHash(hash) === normalizedHash || symbol.toLowerCase() === lowerCaseSymbol
+                )
+
+                return { ...transfer, normalizedHash, token }
+              })
```

Comment:
Just complementing the comment above. 

Change the PendingTransaction interface to have the token, as we always have this information when we create it